components/TARGET_PSA/spm/doc/INTRO.md

```diff
@@ -0,0 +1,117 @@
+
+The SPM and the secure partitions are located in the Secure Processing Environment (SPE), isolating them from the Non-Secure Processing Environment (NSPE), which contains the application firmware, OS kernel and libraries, and other nonsecure hardware resources.
+
+A secure partition is a container for one or more root of trust services, and a platform may have multiple secure partitions. Secure partitions provide the execution environment for security functionality.
+
+Platform hardware, such as the Security Attribution Unit (SAU) and Memory Protection Unit (MPU) in the new ARMv8-M platforms, enforces the separation of partitions. Other platforms may use different mechanisms to provide equivalent isolation for the partitions.
+
+#### PSA levels of isolation
+
+If you are prototyping software or using platforms without SAU or MPU, you can choose to have no isolation between the SPE and NSPE (sometimes referred to as Level 0), though the PSA does not specify this. However, for production software, consider implementing one of the following levels of isolation:
+
+* **Level 1 - SPE isolation** In this level, the SPE is fully isolated from the nonsecure application firmware and hardware.
+* **Level 2 - Root of Trust isolation** In this level, the SPE is fully isolated from the nonsecure application firmware and hardware and the trusted partitions (secure partitions that implement Root of Trust services) are isolated from other secure partitions.
+* **Level 3 - Maximum firmware isolation** In this level, the SPE is fully isolated from the nonsecure application firmware and hardware, and all secure and trusted partitions are individually sandboxed from one another and from the SPM.
+
+### Using secure partitions
+
+Secure partitions are located within the SPE, and must contain at least one set of related security operations (known as a root of trust service) or at least one Interrupt Request (IRQ). You can have multiple root of trust services in a single secure partition.
+
+For a secure partition, you need:
+
+* The **secure partition code**, which must:
+  * Be single threaded.
+  * Be structured as a loop that waits for inputs.
+  * Never exit its loop (considered as a programming error).
+  * Be written in C or 'extern "C"' to avoid C++ name mangling.
+  * Follow PSA IPC rules. Secure partitions communicate with one another using the IPC API defined in [IPC API](https://github.com/ARMmbed/PSA-IPC-doc/blob/master/IPC_revision.md). All IPC messages must eventually be completed [`call psa_end()`]. Note that the SPM does not schedule IPC messages fairly.
+* A **manifest file** in JSON format, that describes the Secure Partition characteristics. The specifications in the manifest file are validated during the build process and at run time.
+
+### Manifest file example
+
+The secure partition manifest file describes the properties of the secure partitions. In this file:
+
+* **entry_point** is the function name of the partition's thread.
```

Comment:
My comments from the initial commit weren't addressed in any way. Not a blocker, but it's weird to have example with some fields documented and some not. You should either document all of them or make the example simpler.

Reply:
fixed in #8873